src/lexer_tok.c

```diff
@@ -0,0 +1,136 @@
+/* ************************************************************************** */
+/*                                                                            */
+/*                                                        :::      ::::::::   */
+/*   lexer_tok.c                                        :+:      :+:    :+:   */
+/*                                                    +:+ +:+         +:+     */
+/*   By: sqiu <sqiu@student.42vienna.com>           +#+  +:+       +#+        */
+/*                                                +#+#+#+#+#+   +#+           */
+/*   Created: 2023/07/02 18:51:59 by sqiu              #+#    #+#             */
+/*   Updated: 2023/07/12 22:46:04 by sqiu             ###   ########.fr       */
+/*                                                                            */
+/* ************************************************************************** */
+
+#include "minishell.h"
+#include "lexer_tok.h"
+
+/**
+ * @brief Create a token.
+ * 
+ * Token contains the token string and its length.
+ * @param token 	Created token.
+ * @param s 		Token string.
+ * @return t_error 	ERR_MALLOC, SUCCESS
+ */
+t_error	ft_create_tok(t_tok *token, char *s)
+{
+	if (!token || !s)
+		return (ERR_EMPTY);
+	token->tok_size = ft_strlen(s);
+	token->tok = ft_realloc (token->tok, token->tok_size + 1, 1);
```

Comment:
Initially this was implemented because the malloced string and its stored value were not carried on to the calling function, unless the string was previously malloced in the calling function. Somehow it works now to malloc in this subfunction and returning the pointer to its calling function; the string value is preserved.

Therefore ft_realloc is replaced with malloc.